pom.xml

```diff
@@ -1,89 +1,636 @@
+        <maven-site-plugin.version>3.12.1</maven-site-plugin.version>
+        <maven-surefire-plugin.version>3.1.2</maven-surefire-plugin.version>
+
+        <!-- 3rd party solver versions -->
+        <java-smt.version>4.0.2</java-smt.version>
+        <boolector.version>3.2.2-g1a89c229</boolector.version>
+        <mathsat.version>5.6.10</mathsat.version>
+        <cvc4.version>1.8-prerelease-2020-06-24-g7825d8f28</cvc4.version>
+        <cvc5.version>1.0.5-g4cb2ab9eb</cvc5.version>
+        <yices2-api.version>4.0.1</yices2-api.version>
+        <yices2.version>2.6.2-396-g194350c1</yices2.version>
+        <z3.version>4.12.2</z3.version>
+
+        <!-- 3rd party library versions -->
+        <antlr.version>4.13.0</antlr.version>
+        <commons-cli.version>1.5.0</commons-cli.version>
+        <guava.version>32.1.2-jre</guava.version>
+        <junit.version>4.13.2</junit.version>
+        <log4j.version>2.20.0</log4j.version>
+        <rsyntaxtextarea.version>3.3.4</rsyntaxtextarea.version>
+
+        <!-- 3rd party plugin versions -->
+        <build-helper-maven-plugin.version>3.4.0</build-helper-maven-plugin.version>
+        <git-commit-id-plugin.version>4.9.10</git-commit-id-plugin.version>
+        <jacoco-maven-plugin.version>0.8.10</jacoco-maven-plugin.version>
+    </properties>
+
+    <modules>
+        <module>dartagnan</module>
+        <module>ui</module>
+        <module>svcomp</module>
+    </modules>
+
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <artifactId>maven-clean-plugin</artifactId>
+                    <version>${maven-clean-plugin.version}</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>${maven-compiler-plugin.version}</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-dependency-plugin</artifactId>
+                    <version>${maven-dependency-plugin.version}</version>
+                    <executions>
+                        <execution>
+                            <id>copy-dependencies</id>
+                            <phase>initialize</phase>
+                            <goals>
+                                <goal>copy-dependencies</goal>
+                            </goals>
+                            <configuration>
+                                <outputDirectory>${project.build.directory}/${dat3m.library.path}</outputDirectory>
+                                <includeTypes>jar</includeTypes>
+                                <includeScope>compile</includeScope>
+                                <excludeGroupIds>com.dat3m</excludeGroupIds>
+                            </configuration>
+                        </execution>
+                        <execution>
+                            <id>copy-binaries</id>
```

Comment:
These are the dependencies from the project itself (i.e., dartagnan.jar), and they cannot be copied before the packaging phase. For example, "mvn test -Dtest="BuildWitnessTest" -pl dartagnan" would fail if this were executed before packaging.